pkg/login/social/generic_oauth.go

```diff
@@ -320,6 +327,27 @@ func (s *SocialGenericOAuth) extractEmail(data *UserInfoJson, userInfoResp []byt
 	return ""
 }
 
+func (s *SocialGenericOAuth) extractRole(data *UserInfoJson, userInfoResp []byte) string {
+	if s.roleAttributePath != "" {
+		role := s.searchJSONForAttr(s.roleAttributePath, userInfoResp)
+		if role != "" {
+			return role
+		}
+	}
+
+	var roles []string
+	if len(data.Roles) > 0 {
+		return data.Roles[0]
+	}
+
+	roles, ok := data.Attributes[s.roleAttributeName]
```

Comment:
`roleAttributeName` shouldn't be needed since role attribute name can be configured using `roleAttributePath`. The idea with using JMESPath is to simplify code and logic for extracting custom attributes in OAuth flow.